packages/fxa-content-server/app/scripts/views/settings.js

```diff
@@ -93,7 +93,9 @@ const View = BaseView.extend({
       ccExpired: !!this._ccExpired,
       escapedCcExpiredLinkAttrs: 'href="/subscriptions" class="alert-link"',
       securityEventsVisible: this.displaySecurityEvents(),
-      showSignOut: !account.isFromSync(),
+      showSignOut:
+        !account.isFromSync() &&
+        !(this.relier.isOAuth() && this.relier.containsOldSyncScope()),
```

Comment:
can you leave a code comment on why this was added? The new logic is a bit cryptic to me as is.